jira/resources.py

```diff
@@ -571,6 +586,7 @@ def __init__(
         self.fields: Issue._IssueFields
         self.id: str
         self.key: str
+        self.raw: Dict[Any, Any] = {}
```

Comment:
```suggestion
```

This shouldn't be here. 

In `Resource.__init__`
```python
        # Explicitly define as None so we know when a resource has actually
        # been loaded
        self.raw: Optional[Dict[str, Any]] = None
```

If you want to fix this you should instead:

```python
        if raw:
            self._parse_raw(raw)
        self.raw: Dict[str, Any]
```

`_parse_raw` will throw an error is `raw` is None so you can guarantee that `self.raw` is a `Dict[str,Any]` after `_parse_raw` is called.

Reply:
Thanks for the heads up, I've already did it the `cast()` way after parsing the raw dict, the type checker is working as expected for me